index.js

```diff
@@ -21,10 +45,11 @@ ListStream.prototype._read = function() {
   var stream = this;
   var q = queue();
   var prefix;
+  var prefixer = new Prefixer(stream.key);
 
-  for (var i = 0; i <= 16; i++) {
-    for (var j = 0; j <= 16; j++) {
-      prefix = stream.key.replace('{prefix}', i.toString(16) + j.toString(16));
+  for (var i = 0; i < prefixer.base; i++) {
```

Comment:
0 to 16, including 16, is 17 possibilities -- and `16.toString(16)` is not something we want:

```
> [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16].map(function(i) {
... return i.toString(16);
... });
[ '0',
  '1',
  '2',
  '3',
  '4',
  '5',
  '6',
  '7',
  '8',
  '9',
  'a',
  'b',
  'c',
  'd',
  'e',
  'f',
  '10' ]
```